scheduler/reconnecting_picker/reconnecting_picker.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package reconnectingpicker
+
+import (
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+type ReconnectingPicker struct {
+	logger log.Logger
+}
+
+func New(logger log.Logger) *ReconnectingPicker {
+	rp := ReconnectingPicker{
+		logger: log.L().Named("reconnecting-picker"),
+	}
+
+	return &rp
+}
+
+func (rp *ReconnectingPicker) PickReconnectingAlloc(ds *structs.DisconnectStrategy, original *structs.Allocation, replacement *structs.Allocation) *structs.Allocation {
+	// Check if the replacement is newer.
+	// Always prefer the replacement if true.
+	replacementIsNewer := replacement.Job.Version > original.Job.Version ||
+		replacement.Job.CreateIndex > original.Job.CreateIndex
+	if replacementIsNewer {
+		rp.logger.Debug("replacement has a newer version, keeping replacement")
+		return replacement
+	}
```

Comment:
```suggestion
	// Check if the replacement is for a newer job version.
	// Always prefer the replacement if true.
	replacementIsNewer := replacement.Job.Version > original.Job.Version ||
		replacement.Job.CreateIndex > original.Job.CreateIndex
	if replacementIsNewer {
		rp.logger.Debug("replacement has a newer job version, keeping replacement")
		return replacement
	}
```

I think the replacement itself would always be newer right? 😅 

So just making this a bit more explicit.